### PR TITLE
feat: Expo camera for pictures

### PR DESCRIPTION
**Pull request by ErikSin**
Introduces [Expo Camera](https://docs.expo.dev/versions/latest/sdk/camera/) and the appropriate android configurations and permissions.

## Expo dependencies added: 
- expo-camera
- expo-screen-orientation
- expo-sensors (accelerometer)

Also introduces a hook `useResetPermissions()` which tracks whether the permissions have been manually turned on in the Android Setting Screen, and syncs the permission state with the Android state.

image

image

### CODE REVIEW

**Review by achou11 — COMMENTED** [review]
some initial comments. I trust that the integration of the module(s) works as expected. have some concerns about the implementation of the hook + screen

File: src/frontend/hooks/useResetPermissions.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import {useEffect, useRef} from 'react';
+import {
+  AppState,
+  Linking,
+  NativeEventSubscription,
+  Permission,
+  PermissionsAndroid,
+} from 'react-native';
+
+type MapeoPermissions = Extract<
```
[the remark this answers is not in the document]
Comment (achou11) [reply src/frontend/hooks/useResetPermissions.ts:10]: Right, but the type reflects a singular value i.e.

`const permission: MapeoPermission = ...`

vs

`const permissions: MapeoPermission[] = ...`

[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/hooks/useResetPermissions.ts:10]: ahh yes, I see what you mean

**Comment by ErikSin** [comment]
Camera was added using previous permission architecture on https://github.com/digidem/react-native-with-mapeo-core-next/pull/49

**Pull request closed without merging**